tests/Parser/parse.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+require_once dirname( __FILE__ ) . '/../../IABot/Parser/parse.php';
+require_once dirname(__FILE__) . '/../../IABot/checkIfDead.php';
+
+class parseTest extends PHPUnit_Framework_TestCase {
+
+	function __construct() {
+		parent::__construct();
+		$api = new API();
+		$this->parser = $this->getMockForAbstractClass( Parser::class, [ $api ] );
+		$this->reflection = new \ReflectionClass( get_class( $this->parser ) );
+	}
+
+	public function testFetchTemplateRegex() {
+		$placeholder = '{{{{templates}}}}';
+		$method = $this->getPrivateMethod( 'fetchTemplateRegex' );
+		$regexOptional = $this->getPrivateProperty( 'templateRegexOptional' );
+		$regexMandatory = $this->getPrivateProperty( 'templateRegexMandatory' );
+
+		$escapedTemplateArray = [ '{{cite web', 'param=foo', 'param2=bar}}' ];
+		$regex = $method->invokeArgs( $this->parser, [ $escapedTemplateArray ] );
+		$testRegex = str_replace( $placeholder, '{{cite web|param=foo|param2=bar}}', $regexOptional );
```

Comment:
I think you misunderstand what the placeholder looks for.  The placeholder will be filled with stuff like "\{\{cite|\{\{cite web|\{\{foo"